symengine/printer.cpp

```diff
@@ -63,14 +63,14 @@ void StrPrinter::bvisit(const Complex &x) {
 
 void StrPrinter::bvisit(const RealDouble &x) {
     std::ostringstream s;
-    s.precision(std::numeric_limits< double >::digits10);
+    s.precision(DBL_DIG);
     s << x.i;
     str_ = s.str();
 }
 
 void StrPrinter::bvisit(const ComplexDouble &x) {
     std::ostringstream s;
-    s.precision(std::numeric_limits< double >::digits10);
+    s.precision(DBL_DIG);
```

Comment:
Double has between 15-17 significant decimal digits (https://en.wikipedia.org/wiki/Double-precision_floating-point_format#IEEE_754_double-precision_binary_floating-point_format:_binary64). So it always has at least 15 digits, but for some numbers it only has 15. That's my understanding.